server/livecheck.py

```diff
@@ -60,10 +95,19 @@ def processCodeData(data):
     # verify certificate
     root_cert_file = makeTempFile(root_cert_text)
     cert_file = makeTempFile(certificate)
+    cert_details = parseCertDetails(cert_file)
 
     cert_verification_result = call(['openssl', 'verify', '-CAfile', root_cert_file, cert_file])
```

Comment:
so this is an important concept here -- don't parse before you verify. If you parse before you verify, you have to write your parsing code particularly defensively, for any number of malicious inputs. We're doing this in a memory-safe language so it's probably okay, but still, as a general rule, first verify the signature, then parse the content. That hugely limits an attacker's ability to mess with us.

Reply:
Oh very interesting. That makes sense, thanks for explaining. Will update